src/libs/AxelarQueryAPI.ts

```diff
@@ -308,4 +314,73 @@ export class AxelarQueryAPI {
       });
     }
   };
+
+  public async getTransferLimit({
+    chainId,
+    denom,
+  }: {
+    denom: string;
+    chainId: string;
+  }): Promise<string> {
+    try {
+      // verify chain params
+      await throwIfInvalidChainIds([chainId], this.environment);
+      const chains = await loadChains({ environment: this.environment });
+      const chain = chains.find((c) => c.id === chainId);
+      if (!chain) throw `Chain ${chainId} not found`;
+
+      // verify asset params
+      if (!this.allAssets) await this._initializeAssets();
+      const asset = this.allAssets.find((asset) => asset.common_key[this.environment] === denom);
+      if (!asset) throw `Asset ${denom} not found`;
+
+      // if chain is on evm module, query the contract directly on the EVM chain
+      // othewise if the chain is in the axelarnet module, query the nexus module
+      return formatUnits(
+        await (chain.module === "evm"
+          ? this.queryEVMTransferLimit(chainId, asset, chain)
+          : this.queryAxelarTransferLimit(chainId, denom)),
+        asset.decimals
+      );
+    } catch (e) {
+      console.error(e);
+      throw `could not fetch transfer limit for ${denom} on ${chainId}`;
+    }
+  }
```

Comment:
Can you narrow down the try-catch block?

The `try-catch` block here is too broad, so it will also intercept an error from the `throwIfInvalidChainIds` function and override the error message.